scipy/linalg/decomp_svd.py

```diff
@@ -416,7 +416,7 @@ def subspace_angles(A, B):
     -----
     This computes the subspace angles according to the formula
     provided in [1]_. For equivalence with MATLAB and Octave behavior,
-    use ``angles[0]``.
+    use ``angles[-1]``.
```

Comment:
The ascending order is just a comment. There was no real order before. The line
`theta = where(mask, mu_arcsin, arccos(clip(sigma, -1., 1.)))`

shows that it was planned in ascending order before because sigma is in descending order and largest sigma leads to smallest arccos (thus ascending order of theta)
The order was destroyed because of wrong ordering in mu_arcsin such that the where() command destroys the results.

Reply:
> There was no real order before.

There must have been some order for the tests to pass (and for you to have to change the value to get them to pass now). Was it just happenstance that they did, then? Or maybe there was some symmetry in the `mu_arcsin` values that made the values from the left look like the values from the right (in the incorrect `master` formulation)?

Reply:
Yes there is a symmetry. In fact this line
`mask = sigma ** 2 >= 0.5`
generates the mask which decides if the angle is calculated by using the arcin or by using the arccos.
If the angle is greater equal than 45deg, the angle is calculated by arccos(sigma).
If the angle is smaller than 45deg, the angle is calculated by arcsin(mu).

This different way of calculation is just because arccos is bad conditioned for small angles. The arcsin is better conditioned for small angles.

The symmetry comes from wrong ordering of mu_arcsin. The largest sin(theta) value belongs to the smallest cos(theta) value. But both, sigma and mu_arcsin were ordered descending.
Therefore, only angles greater than 45deg were returned.

Example:
Right result: [0deg, 20deg, 70deg, 90deg]
Wrong result (bug): [90deg, 70deg, 70deg, 90deg] (<- symmetric and thus no real happenstance).

Reply:
Okay so the advice before to use `angles[0]` was correct then, right? And basically all angles `len(angles) // 2:` were incorrect?

If so, then I'd rather give the angles back in descending order. People might already have (correct) working code that uses `angles[0]`.

Reply:
Okay, because there was a 'mixed' order before we can choose, but probably you are right that some people were only interested in largest angles and used angles[0] for applications that only had angles greater than 45deg. I can make some commits that change the order to descending.